x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_kpis/common/components.tsx

```diff
@@ -5,51 +5,75 @@
  * 2.0.
  */
 
+import { css } from '@emotion/react';
 import type { EuiComboBoxOptionOption, EuiComboBoxProps } from '@elastic/eui';
-import { EuiPanel, EuiComboBox } from '@elastic/eui';
-import styled from 'styled-components';
+import { EuiComboBox, EuiPanel, useEuiTheme } from '@elastic/eui';
 import type { LegacyRef } from 'react';
 import React, { useCallback, useMemo } from 'react';
-import { PANEL_HEIGHT, MOBILE_PANEL_HEIGHT } from './config';
 import { useStackByFields } from './hooks';
 import * as i18n from './translations';
 
 const DEFAULT_WIDTH = 400;
+const DEFAULT_EXPANDED_PANEL_HEIGHT = 300;
+const DEFAULT_MOBILE_EXPANDED_PANEL_HEIGHT = 300;
+const DEFAULT_COLLAPSED_PANEL_HEIGHT = 64; // px
+
+export interface KpiPanelProps {
+  /**
+   * Height to use when the panel is expanded
+   */
+  height?: number | undefined;
+  /**
+   * True if the panel is expanded
+   */
+  toggleStatus: boolean;
+  /**
+   * Data test subject string for testing
+   */
+  ['data-test-subj']: string;
+  /**
+   * Children component
+   */
+  children: React.ReactNode;
+}
+
+export const KpiPanel = ({
+  height,
+  toggleStatus,
+  'data-test-subj': dataTestSubj,
+  children,
+}: KpiPanelProps) => {
+  const { euiTheme } = useEuiTheme();
+
+  return (
+    <EuiPanel
+      hasBorder
+      paddingSize="m"
+      data-test-subj={dataTestSubj}
+      css={css`
+        display: flex;
+        flex-direction: column;
+        position: relative;
+        overflow: hidden;
```

Comment:
I noticed the scrolling in treemap is gone, we should change it to auto / bring back the prop
![image](https://github.com/user-attachments/assets/c1637849-7660-48e4-81e8-011660e01577)